### PR TITLE
avoid taking address of empty vector element

### PR DESCRIPTION
**Pull request by yang-g**

### CODE REVIEW

File: src/cpp/client/secure_credentials.cc [hunk 1]
```diff
@@ -183,7 +183,8 @@ void MetadataCredentialsPluginWrapper::InvokePlugin(
                   0,
                   {{nullptr, nullptr, nullptr, nullptr}}});
   }
-  cb(user_data, &md[0], md.size(),
+  grpc_metadata* md_arr = md.empty() ? NULL : &md[0];
```
Comment (jboeuf) [inline src/cpp/client/secure_credentials.cc]: s/NULL/nullptr

**Comment by jboeuf** [comment]
weird... I though I had fixed this one... Probably didn't push that particular commit... sorry about that... I'll check at home on my laptop if other commits for #3176 are still not merged...

File: src/cpp/client/secure_credentials.cc (hunk 1 above)
Comment (yang-g) [inline src/cpp/client/secure_credentials.cc]: Done.

**Comment by jboeuf** [comment]
Replaced by #3406 which contains another fix that was not checked in.

**Pull request closed without merging**